tensorflow/python/ops/math_ops.py

```diff
@@ -393,8 +396,8 @@ def abs(x, name=None):  # pylint: disable=redefined-builtin
          [6.60492241]])>
 
   Args:
-    x: A `Tensor` or `SparseTensor` of type `float16`, `float32`, `float64`,
-      `int32`, `int64`, `complex64` or `complex128`.
+    x: A `Tensor` or `SparseTensor` of type `bfloat16`, `half`, `float32`, `float64`,
```

Comment:
do we not support float16, complex64 or complex128 for this op?

Reply:
code: Tensorflow Version（2.11）

    import tensorflow as tf
    input = "ds"
    out = tf.math.abs(input)
    print(out)

result: InvalidArgumentError: Value for attr 'T' of string is not in the list of allowed values: bfloat16, half, float, double, int8, int16, int32, int64; NodeDef: {{node Abs}}; Op<name=Abs; signature=x:T -> y:T; attr=T:type,allowed=[DT_BFLOAT16, DT_HALF, DT_FLOAT, DT_DOUBLE, DT_INT8, DT_INT16, DT_INT32, DT_INT64]> [Op:Abs]

As can be seen from the above code, the type of parameter x supports complex types.

Reply:
`float16` and `half` are equivalent.  I believe we usually prefer `float16` in python code.